pkg/handlers/nutanix/mutation/machinedetails/inject.go

```diff
@@ -122,6 +122,24 @@ func (h *nutanixMachineDetailsPatchHandler) Mutate(
 					capxv1.NutanixResourceIdentifier(*nutanixMachineDetailsVar.Project),
 				)
 			}
+			if nutanixMachineDetailsVar.GPUs != nil {
+				spec.GPUs = make(
+					[]capxv1.NutanixGPU,
+					len(nutanixMachineDetailsVar.GPUs),
+				)
+
+				for i, gpu := range nutanixMachineDetailsVar.GPUs {
+					gpuType := gpu.Type
+					spec.GPUs[i] = capxv1.NutanixGPU{
+						Type: gpuType,
+					}
+					if gpuType == capxv1.NutanixGPUIdentifierName {
+						spec.GPUs[i].Name = gpu.Name
+						continue
+					}
+					spec.GPUs[i].DeviceID = gpu.DeviceID
+				}
+			}
```

Comment:
Since its using the upstream type, this can be simplified
```suggestion
			spec.GPUs = make(
				[]capxv1.NutanixGPU,
				len(nutanixMachineDetailsVar.GPUs),
			)
			for i, gpu := range nutanixMachineDetailsVar.GPUs {
				spec.GPUs[i] = gpu
			}
```